force-app/main/default/lwc/geTemplateBuilder/geTemplateBuilder.js

```diff
@@ -1342,51 +1342,60 @@ export default class geTemplateBuilder extends NavigationMixin(LightningElement)
-            }
-
-            this.formLayout.sections = this.formSections;
-            this.formTemplate.batchHeaderFields = this.batchHeaderFields;
-            this.formTemplate.layout = this.formLayout;
-            if (this.disableBatchTableColumnsSubtab && this.mode === NEW) {
-                this.formTemplate.defaultBatchTableColumns = [];
-            } else {
-                this.formTemplate.defaultBatchTableColumns = this.selectedBatchTableColumnOptions;
-            }
-
-            // TODO: Currently hardcoded as we're not providing a way to
-            // create custom migrated field mapping sets yet.
-            this.formTemplate.layout.fieldMappingSetDevName = DEFAULT_FIELD_MAPPING_SET;
-
-            const preppedFormTemplate = {
-                id: this.formTemplateRecordId || null,
-                templateJSON: JSON.stringify(this.formTemplate),
-                name: this.formTemplate.name,
-                description: this.formTemplate.description,
-                formatVersion: FORMAT_VERSION
-            };
-
             try {
-                const recordId = await storeFormTemplate(preppedFormTemplate);
+                const recordId =
+                    await storeFormTemplate(this.preppedFormTemplate());
                 if (recordId) {
                     const toastLabel = this.mode === NEW ?
                         this.CUSTOM_LABELS.geToastTemplateCreateSuccess :
                         this.CUSTOM_LABELS.geToastTemplateUpdateSuccess;
 
-                    const toastMessage = GeLabelService.format(toastLabel, [this.formTemplate.name]);
+                    const toastMessage = GeLabelService.format(
+                        toastLabel, [this.formTemplate.name]);
                     showToast(toastMessage, '', SUCCESS);
                 }
 
                 this.navigateToLandingPage();
             } catch (error) {
-                showToast(this.CUSTOM_LABELS.commonError, this.CUSTOM_LABELS.geToastSaveFailed, ERROR);
+                showToast(this.CUSTOM_LABELS.commonError,
+                    this.CUSTOM_LABELS.geToastSaveFailed, ERROR);
                 this.isLoading = false;
             }
         }
     }
 
+    preppedFormTemplate() {
+        this.buildDefaultFormFields();
+        this.formLayout.sections = this.formSections;
+        this.formTemplate.batchHeaderFields = this.batchHeaderFields;
+        this.formTemplate.layout = this.formLayout;
+        if (this.disableBatchTableColumnsSubtab && this.mode === NEW) {
+            this.formTemplate.defaultBatchTableColumns = [];
+        } else {
+            this.formTemplate.defaultBatchTableColumns
+                = this.selectedBatchTableColumnOptions;
+        }
+
+        // TODO: Currently hardcoded as we're not providing a way to
+        // create custom migrated field mapping sets yet.
+        this.formTemplate.layout.fieldMappingSetDevName
+            = DEFAULT_FIELD_MAPPING_SET;
+       this.stripPermissionsData();
```

Comment:
Does it make more sense to call this at the start of `checkTabsValidity`? That way it's more connected to the actual permission checking logic. It seems sort of... hidden here? Looking at the code though, we may not have the `formTemplate` at that point.

Reply:
It's actually called in this function `preppedFormTemplate`, this function prepares the template to be saved. The `checkTabsValidity` function gets called before we try to save the template.